src/hooks/chatSidebar.ts

```diff
@@ -0,0 +1,85 @@
+import { useCallback } from "react";
+
+import {
+  chatVisibilitySelector,
+  selectedChatSettingsSelector,
+} from "utils/selectors";
+
+import {
+  setPrivateChatTabOpened,
+  setVenueChatTabOpened,
+  setChatSidebarVisibility,
+} from "store/actions/Chat";
+
+import { useSelector } from "./useSelector";
+import { useDispatch } from "./useDispatch";
+import { useNumberOfUnreadChats } from "./privateChats";
+import { useVenueId } from "./useVenueId";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+
+export const useChatSidebarControls = () => {
+  const dispatch = useDispatch();
+  const isExpanded = useSelector(chatVisibilitySelector);
+  const chatSettings = useSelector(selectedChatSettingsSelector);
+
+  const expandSidebar = useCallback(() => {
+    dispatch(setChatSidebarVisibility(true));
+  }, [dispatch]);
+
+  const collapseSidebar = useCallback(() => {
+    dispatch(setChatSidebarVisibility(false));
+  }, [dispatch]);
+
+  const toggleSidebar = useCallback(() => {
+    if (isExpanded) {
+      collapseSidebar();
+    } else {
+      expandSidebar();
+    }
+  }, [expandSidebar, collapseSidebar, isExpanded]);
```

Comment:
Potentially this would be simpler if we implemented it as a 'toggle sidebar' action in redux, and handle the specifics here at the reducer level.